soc/riscv/openisa_rv32m1/soc_offsets.h

```diff
@@ -13,6 +13,7 @@
 
 #ifdef CONFIG_SOC_OPENISA_RV32M1_RI5CY
 
+#ifdef CONFIG_RISCV_SOC_CONTEXT_SAVE
```

Comment:
This config is `default y if SOC_OPENISA_RV32M1_RI5CY`, so the extra ifdef seems redundant? Are you setting `CONFIG_SOC_OPENISA_RV32M1_RI5CY=n` explicitly?

Reply:
I understood it to be an option, defaulting to y, but also allowing n. So one can explicitly disable it if desired, as in less context switch overhead but not getting the benefit of the hardware loop optimization.

I did end up disabling it in a few intermediate configurations when trying to narrow the problem down, to figure out if it was a compiler issue with the hardware loops or the loop registers getting messed up during context switching and it helped there.

Personally I am not attached to having it as an option, just felt it was broken as is thus added this somewhat ugly additional ifdef level

Reply:
Guess I didn't fully answer the question. After fixing the root cause by zero initializing the loop registers in the stack frame I do not set `CONFIG_SOC_OPENISA_RV32M1_RI5CY=n` explicitly :-)

Reply:
@MaureenHelm Sorry I just reread this, i totally referenced the wrong config symbol.
The issue at hand is not `CONFIG_SOC_OPENISA_RV32M1_RI5CY`, but `CONFIG_RISCV_SOC_CONTEXT_SAVE`. It is `default y`, but can be configured to false if desired, to avoid the overhead of the stack operations if one doesn't use the loop optimization feature of the RI5CY core. This commit fixes `CONFIG_RISCV_SOC_CONTEXT_SAVE=n`. I can remove the change if you feel it doesn't need to be supported.